### PR TITLE
use ecto_ch

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR switches to Hex published `ecto_ch` for Ecto ClickHouse adapter.

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/clickhouse_event_v2.ex [hunk 1]
```diff
@@ -7,12 +7,12 @@ defmodule Plausible.ClickhouseEventV2 do
 
   @primary_key false
   schema "events_v2" do
-    field :name, :string
-    field :site_id, Ch.Types.UInt64
+    field :name, Ch, type: "LowCardinality(String)"
```
Comment (ruslandoga) [inline lib/plausible/clickhouse_event_v2.ex:10]: I've switched to `RowBinaryWithNamesAndTypes` for inserts. This allows for a bit of "type-checking" on ClickHouse side. But the types need to be exactly the same as in table definition even if they don't affect the encoded values in any way (like LowCardinality "meta"-type).

File: lib/plausible/clickhouse_session_v2.ex [hunk 2]
```diff
@@ -4,21 +4,46 @@ defmodule Plausible.ClickhouseSessionV2 do
...
-    field :session_id, Ch.Types.UInt64
+    field :site_id, Ch, type: "UInt64"
+    field :user_id, Ch, type: "UInt64"
+    field :session_id, Ch, type: "UInt64"
 
     field :start, :naive_datetime
-    field :duration, Ch.Types.UInt32
-    field :is_bounce, :boolean
+    field :duration, Ch, type: "UInt32"
+    field :is_bounce, BoolUInt8
```
Comment (ruslandoga) [inline lib/plausible/clickhouse_session_v2.ex:41]: Need a custom type here to insert bools into `UInt8` column.

File: lib/plausible/clickhouse_session_v2.ex [hunk 3]
```diff
@@ -31,16 +56,16 @@ defmodule Plausible.ClickhouseSessionV2 do
     field :referrer, :string
     field :referrer_source, :string
 
-    field :country_code, Ch.Types.FixedString, size: 2
-    field :subdivision1_code, :string
-    field :subdivision2_code, :string
-    field :city_geoname_id, Ch.Types.UInt32
+    field :country_code, Ch, type: "LowCardinality(FixedString(2))"
```
Comment (ruslandoga) [inline lib/plausible/clickhouse_session_v2.ex:59]: Btw `LowCardinality(FixedString(2))` seems a bit extra. There is a warning in ClickHouse now when `LowCardinality` is used on types less than 8 bytes.

https://clickhouse.com/docs/en/operations/settings/settings#allow_suspicious_low_cardinality_types

File: lib/plausible/clickhouse_session_v2.ex [hunk 4]
```diff
@@ -4,21 +4,46 @@ defmodule Plausible.ClickhouseSessionV2 do
...
 
+  defmodule BoolUInt8 do
+    @moduledoc """
+    Custom type to cast Bool as UInt8
+    """
+
+    use Ecto.Type
+
+    @impl true
+    def type, do: :u8
```
Comment (ruslandoga) [inline lib/plausible/clickhouse_session_v2.ex:15]: - [ ] `Ecto.ParameterizedTypes.init(Ch, type: "UInt8")`

Comment (ruslandoga) [reply lib/plausible/clickhouse_session_v2.ex:15]:
```suggestion
    u8 = Ecto.ParameterizedTypes.init(Ch, type: "UInt8")
    
    @impl true
    def type, do: unquote(u8)
```

**Review by ukutaht — APPROVED** [review]

**Review by aerosol — APPROVED** [review]

**Comment by aerosol** [comment]
Could you rebase so that the lockfile conflicts are resolved?

**Comment by ruslandoga** [comment]
Sure! Will do now.

**Comment by ruslandoga** [comment]
Done, and I've also updated `ecto_ch` and `ch` to latest versions.

**Pull request merged**